src/runtime-rs/crates/hypervisor/src/ch/inner_device.rs

```diff
@@ -161,32 +152,53 @@ impl CloudHypervisorInner {
         Ok(())
     }
 
-    pub(crate) async fn get_shared_fs_devices(&mut self) -> Result<Option<Vec<FsConfig>>> {
-        let pending_root_devices = self.pending_devices.take();
-
-        let mut root_devices = Vec::<FsConfig>::new();
+    pub(crate) async fn get_shared_devices(
+        &mut self,
+    ) -> Result<(Option<Vec<FsConfig>>, Option<Vec<NetConfig>>)> {
+        let mut shared_fs_devices = Vec::<FsConfig>::new();
+        let mut network_devices = Vec::<NetConfig>::new();
+
+        while let Some(dev) = self.pending_devices.pop() {
+            match dev {
+                DeviceType::ShareFs(dev) => {
+                    let settings = ShareFsSettings::new(dev.config, self.vm_path.clone());
+
+                    let fs_cfg = FsConfig::try_from(settings)?;
+
+                    shared_fs_devices.push(fs_cfg);
+                }
+                DeviceType::Network(net_device) => {
+                    let net_config = NetConfig::try_from(net_device.config)?;
+                    network_devices.push(net_config);
+                }
+                _ => continue,
+            }
+        }
 
-        if let Some(devices) = pending_root_devices {
-            for dev in devices {
-                match dev {
-                    DeviceType::ShareFs(dev) => {
-                        let settings = ShareFsSettings::new(dev.config, self.vm_path.clone());
+        Ok((Some(shared_fs_devices), Some(network_devices)))
+    }
+}
 
-                        let fs_cfg = FsConfig::try_from(settings)?;
+impl TryFrom<NetworkConfig> for NetConfig {
+    type Error = anyhow::Error;
 
-                        root_devices.push(fs_cfg);
-                    }
-                    _ => continue,
-                };
-            }
+    fn try_from(cfg: NetworkConfig) -> Result<Self, Self::Error> {
+        let mut net_config = NetConfig {
```

Comment:
- If set, should / can `NetworkConfig.virt_iface_name` be used for `net_config.id`?

Reply:
This is strictly not required, as the agent relies on the mac address solely and not on the interface name for updating the interfaces later within the VM. But no harm updating this.